lib/tasks/instrumentation_generator/templates/dependency_detection.tt

```diff
@@ -3,7 +3,7 @@
 # frozen_string_literal: true
 
 DependencyDetection.defer do
-  named :<%= @name.match?(/\-|\_/) ? "'#{@snake_name}'" : @name.downcase %>
+  named :<%= (@name == @snake_name ? @name : @snake_name).downcase %>
```

Comment:
the more I look at this, maybe it just needs to be `<%= @snake_name %>`? 
* we'll have an `@snake_name` assigned every time, that will either be equal to `@name` or be the `@name` with hyphens transformed into underscores
* `snake_name()` calls `downcase`, so we shouldn't have to here

Reply:
Great call. Once we're out of the business of using quotes or not, we can just use `@snake_name`. Updated [the commit]([2507052](https://github.com/newrelic/newrelic-ruby-agent/pull/2918/commits/2507052f091da56e21ecea71ec6d918d4b942584)). Will update the PR description accordingly.